src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx

```diff
@@ -149,22 +144,11 @@ function BarplotViz(props: Props) {
       const xAxisVariable = findVariable(vizConfig.xAxisVariable);
       const overlayVariable = findVariable(vizConfig.overlayVariable);
 
-      // check variable inputs and add densityplot
+      // check variable inputs: this is necessary to prevent from data post
       if (vizConfig.xAxisVariable == null || xAxisVariable == null)
-        return Promise.reject(new Error('Please choose a X-axis variable'));
-      // overlay
-      else if (
-        vizConfig.overlayVariable != null &&
-        overlayVariable != null &&
-        !isTableVariable(overlayVariable)
-      )
-        return Promise.reject(
-          new Error(
-            `'${overlayVariable.displayName}' is not suitable for this plot. Only categorical, binary, or ordinal type is allowed for the Overlay variable.`
-          )
-        );
+        return Promise.reject();
```

Comment:
I am very wary of this. I think it would be better to just return `undefined`. This applies to all such cases.

As a side note, I would prefer to see errors thrown in async functions, rather than returning a rejected Promise.

Reply:
@dmfalke thank you for your review and comments 👍 Actually I was gauging to use either empty `promise.reject()` or `undefined` as both worked fine for the purpose of not showing error message. And it makes sense to me for your second point as well.

changed to use `undefined` for both scatter and barplot vizs

Reply:
If the works at scatterplot and barplot vizs are acceptable, then I can probably do the same thing for other vizs like histogram and mosaic plot ones on the behalf of Bob and Connor.

Reply:
I think this looks good. Thanks!

Reply:
@dmfalke Thank you for your review. Then I will work for other vizs as well as it may not be that difficult :)